metabase/metabase
# Fix form reinitialization with formik

Epic https://github.com/metabase/metabase/issues/26178

Currently the submit button doesn't show `Success` message when `enableReinitialize` is passed to `Formik`. This happens because `Formik` resets the internal `status` field used for submit status tracking. This PR fixes the issue by moving the submit state to its own context.

How to test:
- Account -> Profile
- Change any field and click `Update`
- Make sure the button displays `Success` and becomes disabled after submit

<img width="622" alt="Screenshot 2022-11-08 at 15 47 50" src="https://user-images.githubusercontent.com/8542534/200581523-8ad96202-5a62-401a-8e8a-8d17b80139bb.png">
